src/openms/include/OpenMS/KERNEL/MSExperiment.h

```diff
@@ -367,20 +367,42 @@ namespace OpenMS
         if (it.getRT() != t) 
         {
           t = (float)it.getRT();
-          rt.emplace_back(t);
-          mz.resize(mz.size() + 1); 
-          rt.resize(rt.size() + 1);
-          intensity.resize(intensity.size() + 1);
+          rt.push_back(t);
         }
         mz.back().push_back((float)it->getMZ());
-        intensity.back().emplace_back(it->getIntensity());
+        intensity.back().push_back(it->getIntensity());
       }
     }
 
-    // for fast pyOpenMS access to MS1 peak data in format: [rt, mz, intensity]
+    // for fast pyOpenMS access to MS1 peak data in format: [rt, [mz, intensity, ion mobility]]
     void get2DPeakData(CoordinateType min_rt, CoordinateType max_rt, CoordinateType min_mz, CoordinateType max_mz, 
       std::vector<float>& rt, 
-      std::vector<float>& mz, 
+      std::vector<std::vector<float>>& mz,
+      std::vector<std::vector<float>>& intensity, 
+      std::vector<std::vector<float>>& ion) const
+    {
+      float t = -1.0;
+      for (auto it = areaBeginConst(min_rt, max_rt, min_mz, max_mz); it != areaEndConst(); ++it)
+      {
+        if (it.getRT() != t)
+        {
+          t = (float)it.getRT();
+          rt.push_back(t);
+        }
+        mz.back().push_back((float)it->getMZ());
+        intensity.back().push_back(it->getIntensity());
+        ion.back().push_back(it.getDriftTime());
```

Comment:
Note: this is the drift time of the spectrum not of the peak.
If this makes sense depends a lot on how the spectra were converted.
e.g. in case you have a float data array "Ion Mobility" one needs to take the value from there

Reply:
Then I think this PR should be parked until this is properly implemented and abstracted from the developer.
It shouldn't be, that every time you want an ion mobility for a peak you need to consider multiple very hidden places.

Reply:
I pinged our IM guys. Maybe we just need some additional columns to accommodate for this.

Reply:
I think you could do the same in C++.
first determine the IM float data array with https://github.com/OpenMS/OpenMS/blob/develop/src/openms/include/OpenMS/KERNEL/MSSpectrum.h#L561

Here: https://github.com/OpenMS/OpenMS/pull/6941/files#diff-69ce71d959bba39e51afd6283c5159bbb6e41124a75f9d1ae4e49fe481f158d4R394 you can access the current spectrum and peak index with:
auto spectrum_index = it.getPeakIndex().spectrum; 
auto peak_index = it.getPeakIndex().peak; 
So you could take this peak_index to extract the IM value from the IM data float array. But: at first glance, it looks like there is no getter in AreaIterator to access the IM data array. @jpfeuffer would it make sense to add an it.getSpectrum() that returns a const MSSpectrum&? then this could be used to access the IM data

Reply:
Sure. But you can also do it internally if you don't want to expose spectrum access.
The bigger problem are ambiguous annotations though.